extensions/amp-viewer-integration/0.1/amp-viewer-integration.js

```diff
@@ -56,31 +57,76 @@ export class AmpViewerIntegration {
     dev().fine(TAG, 'handshake init()');
     const viewer = viewerForDoc(this.win.document);
     this.unconfirmedViewerOrigin_ = viewer.getParam('origin');
-    if (!this.unconfirmedViewerOrigin_) {
-      dev().fine(TAG, 'Viewer origin not specified.');
-      return null;
+    this.unconfirmedViewerOrigin_ = ''; // mocking what would happen with a native app.
+
+    const win = this.win;
+    const unconfirmedViewerOrigin = this.unconfirmedViewerOrigin_;
+    if (this.unconfirmedViewerOrigin_) {
+      class WindowPortEmulator {
```

Comment:
Should be moved to the top-level class. Probably in the `messaging.js` file or a separate file - either or.